src/adapters/sqlite/makeDispatcher/node/Database.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+const fs = require(`fs`)
+const SQliteDatabase = require('better-sqlite3')
+
+class Database {
+  instance: SQliteDatabase = undefined
+
+  path: string
+
+  constructor(path: string = ':memory:'): void {
+    this.path = path
+    // this.instance = new SQliteDatabase(path);
+    this.open()
+  }
+
+  open = () => {
+    let { path } = this
+    if (path === 'file::memory:' || path.indexOf('?mode=memory') >= 0) {
+      path = ':memory:'
+    }
+
+    try {
+      // eslint-disable-next-line no-console
+      this.instance = new SQliteDatabase(path, { verboze: console.log })
+    } catch (error) {
+      throw new Error(`Failed to open the database. - ${error.message}`)
+    }
+
+    if (!this.instance || !this.instance.open) {
+      throw new Error('Failed to open the database.')
+    }
+  }
+
+  inTransaction = (executeBlock: () => void) => this.instance.transaction(executeBlock)()
+
+  execute = (query: string, args: any[] = []) => this.instance.prepare(query).run(args)
+
+  executeStatements = (queries: string) => this.instance.exec(queries)
+
+  queryRaw = (query: string, args: any[] = []) => {
+    let results = []
+    const stmt = this.instance.prepare(query)
+    if (stmt.get(args)) {
+      results = stmt.all(args)
+    }
+    return results
+  }
+
+  count = (query: string, args: any[] = []) => {
+    const results = this.instance.prepare(query).all(args)
+
+    if (results.length === 0) {
+      throw new Error('Invalid count query, can`t find next() on the result')
+    }
+
+    const result = results[0]
+
+    if (result.count === undefined) {
+      throw new Error('Invalid count query, can`t find `count` column')
+    }
+
+    return Number.parseInt(result.count, 10)
+  }
+
+  getUserVersion = (): number => {
+    return this.instance.pragma('user_version', {
+      simple: true,
+    })
+  }
+
+  get userVersion(): number {
+    return this.getUserVersion()
+  }
+
+  set userVersion(version: number): number {
+    this.instance.pragma(`user_version = ${version}`)
+    return this.getUserVersion()
+  }
+
+  unsafeDestroyEverything = () => {
+    // Deleting files by default because it seems simpler, more reliable
+    // And we have a weird problem with sqlite code 6 (database busy) in sync mode
+    // But sadly this won't work for in-memory (shared) databases, so in those cases,
+    // drop all tables, indexes, and reset user version to 0
+
+    if (this.isInMemoryDatabase()) {
+      const results = this.queryRaw(`SELECT * FROM sqlite_master WHERE type = 'table'`)
```

Comment:
this should be in the transaction

Reply:
Mmm... this also means we should have a test failing here... will work on it...

Reply:
ideally, yeah, but it might be hard to write a test for ;)